File: pattogo.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>


static void gotoxy(int x,int y)

{
	int i;
	for(i=0; i<y; i++) printf("\n");
	for(i=0; i<x; i++) printf(" ");
	printf("o\n");
}

void usleep(int);

int main(void)
{

int egyx=1;
int egyy=-1;

int i;

int x=10;
int y=20;

int ty[23];
int tx[80];

for(i=0;i<23; i++)
	ty[i]=1;

ty[1]=-1;
ty[22]=-1;

for(i=0; i<79; i++)
	tx[i]=1;

tx[1]=-1;
tx[79]=-1;

for (;;)
{
for(i=0; i<36; i++)
	printf("_");

printf("x=%2d",x);
printf("y=%2d",y);

for(i=0; i<36; i++)
	printf("_");


(void)gotoxy(x,y);

x+=egyx;
y+=egyy;

egyx*=tx[x];
egyy*=ty[y];

usleep(20000);
(void)system("clear");
}
}

File: pattogo.py
import os
from time import sleep

def cls():
    os.system('cls' if os.name=='nt' else 'clear')

def gotoxy(x,y):
    for i in range(y):
        print()
    for i in range(x):
        print(" ", end="")
    print("o")

egyx = 1
egyy = -1
x = 10
y = 20
ty = []
tx = []

for i in range(23):
    ty.append(1)

ty[1] = -1
ty[22] = -1

for i in range(80):
    tx.append(1)

tx[1] = -1
tx[79] = -1


while True:
    for i in range(37):
        print("_", end="")

    print("x=", x, end="")
    print("y=", y, end="")

    for i in range(37):
        print("_", end="")

    gotoxy(x,y)

    x+=egyx
    y+=egyy

    egyx*=tx[x]
    egyy*=ty[y]

    sleep(0.5)
    cls()
